mlflow/data/artifact_dataset_sources.py

```diff
@@ -137,16 +137,16 @@ def _can_resolve(raw_source: Any):
         def _resolve(cls, raw_source: Any) -> DatasetForArtifactRepoSourceType:
             return cls(str(raw_source))
 
-        def _to_dict(self) -> Dict[str, str]:
```

Comment:
str -> str is too restrictive. As long as the dict is JSON-compatible, it's fine :)